### PR TITLE
Feature: getters and setters

### PR DESCRIPTION
**Pull request by ruttam**

### CODE REVIEW

File: src/types/include/channel.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+class Channel
+{
+	private:
+		PortMapping portMapping[100];
+    	monotonic_buffer_resource portMappingSrc{std::data(portMapping),
+                                               std::size(portMapping)};
+		identifier_t channelIdentifier;					/* required */
+		std::optional<name_t> channelName;			/* optional */
+		PortMapping source;									/* required */
+		vector<PortMapping> destinations; 		/* required */
```
Comment (TeiturMcSwagger) [inline src/types/include/channel.hpp]: maybe fix indentation of comments

File: src/types/include/module_hm_table.hpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class ModuleHMTable
+{
+    private:
+      ModuleErrorAction moduleErrorAction[100];
+      monotonic_buffer_resource moduleErrorActionSrc{std::data(moduleErrorAction),
+                                                    std::size(moduleErrorAction)};
+      identifier_t stateIdentifier;       /* required */
+      std::string description;            /* required */
+      vector<ModuleErrorAction> actions;   /* required */
```
Comment (TeiturMcSwagger) [inline src/types/include/module_hm_table.hpp]: maybe change indentation

File: src/types/include/partition.hpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      Partition(identifier_t id, PROCESSOR_CORE_ID_TYPE afinity, name_t name,
+                decOrHex_t duration, decOrHex_t period,
+                std::initializer_list<MemoryRegion> mem,
+                std::initializer_list<QueuingPort> queuing,
+                std::initializer_list<SamplingPort> sampling,
+                std::initializer_list<Process> proc):
+        partitionIdentifier(id), affinity(affinity), partitionName(name),
+        duration(duration), period(period), memoryRegions(mem, &memoryRegionSrc),
+        queuePorts(queuing, &queuingPortSrc), samplePorts(sampling, &samplingPortSrc),
+        processes(proc, &processSrc) {}
```
Comment (TeiturMcSwagger) [inline src/types/include/partition.hpp:61]: Maybe add indentation

File: src/types/include/port.hpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#include "apex_types.h"
+#include "port_mapping.hpp"
+
+
+class Port
+{
+  private:
+    name_t portName;                /* required */
+    APEX_INTEGER maxMessageSize;             /* required */
+    PORT_DIRECTION_TYPE direction;  /* required */
```
Comment (TeiturMcSwagger) [inline src/types/include/port.hpp]: Maybe add indentation

**Review by TeiturMcSwagger — CHANGES_REQUESTED** [review]

File: src/types/include/partition.hpp (hunk 3 above)
Comment (ruttam) [reply src/types/include/partition.hpp:61]: The indentation of parameter list or the attribute assignments? For me it looks fine..

**Pull request merged**